### ISSUE TITLE
Strict date parsing due to XMLSchema compliance

### ISSUE THREAD

**Issue by elad-shaked** [eclipse/rdf4j#2436]
Parsing this literal `"0000"^^<http://www.w3.org/2001/XMLSchema#gYear>` as a `SimpleLiteral` and then trying to get the calendar value with `SimpleLiteral.calendarValue()` fails as gregorian year "0000" is invalid in XMLSchema1.0 However, this was changes and is valid in XMLSchema1.1
RDF4J uses Xerces. If what is written [here](https://stackoverflow.com/a/53452378/2261244) is correct then I am not hopeful of XML Schema 1.1 support. WDYT?

**Comment by jeenbroekstra** [comment]
RDF4J doesn't explicitly use Xerces - the processing relies on your JAXP implementation, which by default is the Xerces implementation that comes bundled with the JDK. You can inject a different processor with XSD 1.1 support if you need it.

However, having said that, in this case that likely won't help: RDF4J validates calendar datatypes by converting their lexical value to a `javax.xml.datatype.XMLGregorianCalendar`, which is explicitly documented in the JDK javadoc as supporting XML Schema 1.0. I did a quick test with a version of Xerces that supposedly supports XML Schema 1.1, and it still gave me a parse error on this literal, because the `XMLGregorianCalendarImpl` class it provides does not accept the value. 

XML Schema 1.1 datatype support seems to be poorly represented in the Java ecosystem. I'm open to suggestions. Unless someone can point me to a JAXP implementation that accepts XML Schema 1.1 datatypes in its `XMLGregorianCalendar` implementation,  we should probably change our validation code to no longer use `javax.xml.datatype.XMLGregorianCalendar` / `javax.xml.datatype.DatatypeFactory`. I'm not immediately sure what we would replace it with though.

**Comment by elad-shaked** [comment]
Yeah, seems like too much work for a small thing. Im fine with closing this ticket

**Comment by jeenbroekstra** [comment]
I'll keep this open for now if you don't mind. I can't commit to any timeline on this but we should try and see if we can somehow enable full XMLSchema 1.1 support.

**Comment by barthanssens** [comment]
Looks like there are two versions of Xerces, one with and one without support for XMLSchema 1.1 (http://xerces.apache.org/mirrors.cgi).

Unfortunately OpenJDK (at least till v15) uses the version without support for XMLSchema 1.1 (see also https://bugs.openjdk.java.net/browse/JDK-8242470)
